agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -949,12 +930,15 @@ func (dg *dockerGoClient) KnownVersions() []dockerclient.DockerVersion {
 	return dg.clientFactory.FindKnownAPIVersions()
 }
 
-func (dg *dockerGoClient) Version() (string, error) {
+func (dg *dockerGoClient) Version(ctx context.Context, timeout time.Duration) (string, error) {
+	derivedCtx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
 	client, err := dg.dockerClient()
 	if err != nil {
 		return "", err
 	}
-	info, err := client.Version()
+	info, err := client.VersionWithContext(derivedCtx)
```

Comment:
Why are we not listening to the `ctx.Done` channel like the other Docker APIs?

Reply:
Other APIs require special error handling on timeouts. They all require the output from Docker APIs to be massaged in some way and are invoked in an internal async method. `Version()` doesn't need any such semantics.